### DISCUSSION TITLE
Advanced Foundry - lesson 19 - Create the fuzz test handler part 1 Problem with --invariant-runs Configuration in Forge

### DISCUSSION THREAD

**Discussion by 0xivanalexander** [Cyfrin/foundry-full-course-cu#2849] [Q&A] [answered]
I am running an invariant test in Forge where I expect the test to perform 12,000 fuzz runs, but the test execution finishes much faster than expected, taking only around 20 milliseconds. This suggests that the --fuzz-runs configuration might not be functioning correctly.

Code Setup
The invariant test checks two properties:

The total supply of my Decentralized Stable Coin (DSC) should be less than or equal to the total value of collateral in the protocol.
Getter view functions should not revert.
Here’s my code for InvariantsTest.sol:

```
pragma solidity 0.8.20;

import {Test, console} from "forge-std/Test.sol";

import {StdInvariant} from "forge-std/StdInvariant.sol";
import {DeployDSC} from "../script/DeployDSC.s.sol";
import {DSCEngine} from "src/DSCEngine.sol";
import {DecentralizedStableCoin} from "src/DecentralizedStableCoin.sol";
import {HelperConfig} from "script/HelperConfig.s.sol";
import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";

contract InvariantsTest is StdInvariant, Test {
    DeployDSC deployer;
    DSCEngine dsce;
    DecentralizedStableCoin dsc;
    HelperConfig config;
    address weth;
    address wbtc;
```

    function setUp() external {
        deployer = new DeployDSC();
        (dsc, dsce, config) = deployer.run();
        (,,,, weth, wbtc,) = config.activeNetworkConfig();
        targetContract(address(dsce));
    }

    function setUp() external {
        deployer = new DeployDSC();
        (dsc, dsce, config) = deployer.run();
        (,,,, weth, wbtc,) = config.activeNetworkConfig();
        targetContract(address(dsce));
    }

    function testinvarient_protocolMustHaveMoreValueThanTotalSupply() public view {
        uint256 totalSupply = dsc.totalSupply();
        uint256 totalWethDeposited = IERC20(weth).balanceOf(address(dsce));
        uint256 totalWbtcDeposited = IERC20(wbtc).balanceOf(address(dsce));

        uint256 wethValue = dsce.getUsdValue(weth, totalWethDeposited);
        uint256 wbtcValue = dsce.getUsdValue(wbtc, totalWbtcDeposited);

        console.log("weth value", wethValue);
        console.log("wbtc value", wbtcValue);
        console.log("total supply", totalSupply);
        assert(wethValue + wbtcValue >= totalSupply);
    }
}

Configuration (foundry.toml)
In my foundry.toml, I’ve set up the following [fuzz] and [invariant] sections to specify fuzzing parameters:
'''

>  [profile.default]
> src = "src"
> out = "out"
> libs = ["lib"]
> remappings = [ "@chainlink/contracts/=lib/chainlink-brownie-contracts/contracts/", "@openzeppelin/contracts=lib/openzeppelin-contracts/contracts"]
> 
> [fuzz]
> runs = 12000
> depth = 128
> fail_on_revert = false
> 
> [invariant]
> runs = 12000
> depth = 128
> fail_on_revert = false
```

Command Execution
I’m running the test with the following commands:
```
forge test --match-path test/OpenInvariantsTest.t.sol --fuzz-runs 12000 --show-progress --gas-report
forge test --match-test testinvarient_protocolMustHaveMoreValueThanTotalSupply --fuzz-runs 12000 --show-progress --gas-report
```
Observed Issue
Despite setting --fuzz-runs 12000, the test completes in a fraction of a second, indicating it’s not actually performing the full number of fuzz runs. Here’s a summary of the output:

`Suite result: ok. 1 passed; 0 failed; 0 skipped; finished in 19.12ms (78.04µs CPU time)
`It seems the fuzzing configuration may not be recognized, or Forge isn’t applying the expected 12,000 runs.

Troubleshooting Steps Taken
Checked [fuzz] and [invariant] Sections in foundry.toml: Confirmed the runs setting is set to 12,000 in both sections.
Ran forge clean to ensure no old artifacts were affecting the test.
Verified I’m on the latest Forge version.

**Comment by 0xivanalexander** [comment, marked as answer]
Pfff what an easy mistake this was, I took a 2 hour break, came back asked chatgpt again but formated my question better and the solution was this...

 Correct the Function Name:
Why: Foundry identifies invariant tests based on their naming convention. Specifically, functions intended as invariant tests should start with the prefix invariant_.

How: Rename your function from invarient_protocolMustHaveMoreValueThanTotalSupply to invariant_protocolMustHaveMoreValueThanTotalSupply.

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out.
